user/best-practices-security.md

```diff
@@ -0,0 +1,62 @@
+
+
+## Recommendations on how to avoid leaking secrets to build logs
+Despite our best efforts, there are however many ways in which secure information can accidentally be exposed. These vary according to what tools you are using and what settings you have enabled. Some things to look out for are:
+
+* settings which duplicate commands to standard out, such as `set -x` or `set -v` in your bash scripts
+* displaying environment variables, by running `env` or `printenv`
+* printing secrets within the code, for example `echo $SECRET_KEY`
+* using tools that prints out secrets on error, such as `php -i`
+* git commands like git fetch or git push may expose GitHub tokens if the command fails
+* mistakes in string escaping 
+* settings which increase command verbosity
+* testing tools or plugins that may expose secrets while operating
+
+
+Preventing commands from displaying any output is one way to avoid accidentally displaying any secure information. If there is a particular command that is using secure information you can redirect its output to `/dev/null` to make sure it does not accidentally publish anything.
+
+`git push url-with-secret >/dev/null 2>&1`
+
+## If you think that you might have exposed secure information
+
+As an initial step, it’s possible to delete logs containing any secure information by clicking the “Remove log” button in our UI:
+
+![remove log button](/images/remove-log.png "remove log button")
+
+If you discover a leak in one of your build logs it’s essential that you revoke the leaked token or environment variable, and update any build scripts or commands that caused the leak.
+
+## Alternative methods to delete logs that may have exposed secure information
+
+### Using the Travis CI API
+
+Using the log endpoint of the Travis CI API you can delete build logs too:
+
+[For public repositories](https://developer.travis-ci.org/resource/log#delete)
+
+[For private repositories](https://developer.travis-ci.com/resource/log#delete)
+
+You can use our API to find the the Job ID or displayed in the build log after expanding "Build system information". 
+
+For this you would need to have the job IDs belonging to your organization. Those you would get from the [`build` end point](https://developer.travis-ci.com/resource/build#standard-representation).
+
+### Using the Travis Command Line Tool
+
+For instructions on how to delete build logs using the CLI please check https://github.com/travis-ci/travis.rb#logs 
+ 
+## Rotate tokens and secrets periodically
+As a general measure, tokens and secrets should be rotated regularly. GitHub OAuth tokens can be found in your [Developer Settings](https://github.com/settings/developers) on the GitHub site. Please regularly rotate credentials for other third-party services as well. 
```

Comment:
general measure is vague. We could just say "Rotate your tokens and secrets regularly: